core/lib/dal/src/blocks_dal.rs

```diff
@@ -2334,6 +2355,84 @@ impl BlocksDal<'_, '_> {
 
         Ok(results.into_iter().map(L::from).collect())
     }
+
+    pub async fn get_last_l2_block_bloom(&mut self) -> DalResult<Option<Bloom>> {
+        let row = sqlx::query!(
+            r#"
+            SELECT
+                logs_bloom
+            FROM
+                miniblocks
+            ORDER BY
+                number DESC
+            LIMIT
+                1
+            "#,
+        )
+        .instrument("get_last_l2_block_bloom")
+        .fetch_optional(self.storage)
+        .await?;
+
+        Ok(row
+            .and_then(|row| row.logs_bloom)
+            .map(|b| Bloom::from_slice(&b)))
+    }
+
+    pub async fn get_max_l2_block_without_bloom(&mut self) -> DalResult<Option<L2BlockNumber>> {
+        let row = sqlx::query!(
+            r#"
+            SELECT
+                MAX(number) AS "max?"
+            FROM
+                miniblocks
+            WHERE
+                logs_bloom IS NULL
+            "#,
+        )
+        .instrument("get_max_l2_block_without_bloom")
+        .fetch_one(self.storage)
+        .await?;
+
+        Ok(row.max.map(|n| L2BlockNumber(n as u32)))
+    }
+
+    pub async fn range_update_logs_bloom(
+        &mut self,
+        from_l2_block: L2BlockNumber,
+        to_l2_block: L2BlockNumber,
+        blooms: Vec<Bloom>,
+    ) -> DalResult<()> {
+        let numbers: Vec<_> = (i64::from(from_l2_block.0)..=i64::from(to_l2_block.0)).collect();
+        assert_eq!(
+            numbers.len(),
+            blooms.len(),
+            "range_update_logs_bloom: inconsistent array lengths"
+        );
+
+        let blooms: Vec<_> = blooms.into_iter().map(|b| b.0.to_vec()).collect();
```

Comment:
IIUC, it's possible to avoid allocations here by converting each `Bloom` to `&[u8]` and specifying `&blooms as &[&[u8]]` in the query below.